### ISSUE TITLE
Add Delete functionality

### ISSUE THREAD

**Issue by danielamiao** [momentohq/momento-cli#145]
Our backend server now supports delete, this functionality needs to be added into the CLI.

Need to update the rust sdk first https://github.com/momentohq/client-sdk-rust/issues/49

**Comment by danielamiao** [comment]
I was using our CLI the other day and realized sometimes I accidentally early return so:
`momento cache delete --cachename myCache --key myKey` might turn into `momento cache delete --cachename myCache` which would be terrible because now my entire cache is gone. We can:
1. use `delete-item` command instead of `delete` for a single item deletion OR
2. add a confirmation prompt to delete cache.

**Comment by danielamiao** [comment]
To clarify, the full CLI command should be:

```
momento set
momento get
momento delete-item
momento create-cache
momento list-caches
momento delete-cache

```

**Comment by kvcache** [comment]
```
momento cache get key
momento cache set key value --cache mycoolcache
momento cache delete key
momento create cache mycoolcache
momento list caches
momento drop cache mycoolcache
```
This is what I think it should be instead (with example target state valid invocations included, not just command names).

**Comment by danielamiao** [comment]
@ksmotiv8 what do you think about the proposal above from @kvcache ?

**Comment by ksmotiv8** [comment]
I like it - I still believe that the word `cache` is almost always superfluous. 


momento get key
momento set key value --cache mycoolcache
momento delete key
momento create cache mycoolcache
momento list caches
momento drop cache mycoolcache

**Comment by cprice404** [comment]
i mostly agree with kenny's suggestion - and I think that if we believe a noun is necessary after `create`, `list`, and `drop`, then it is also necessary before `get` and `set`.

Why do we think that noun is necessary for one of the two use cases but not the other?  And don't we want to make sure we are forward compatible with new nouns coming into our lexicon?

**Comment by ksmotiv8** [comment]
In SQL land, we say `insert into table-name` - but we say `drop table table-name` , `show tables` and `create table table-name`

**Comment by kvcache** [comment]
sql has 1 noun-class into which one may `insert` though. Chris is calling attention to the extension of our object types such that the nature of interactions may not be reasonably overlapping between them.

SQL is internally consistent with Chris's feedback too: You can drop more than 1 noun-class of named entity, so you disambiguate by saying which noun-class of thing you're dropping; `table`  in your example, but `procedure`, `index` and `function` are among the other things one might `drop`.

If one were able to insert rows into functions in sql, you would see the language more like `insert into [table|function] name VALUES [...]` but that doesn't make a lot of sense to a database. Though one can insert into views, views are table-like. If we are to have nouns that a user can interact with for which the verb set does not alias (ie, behaviorally different noun classes), we need to have an identifier to group and clarify the valid functionality per behavioral-noun-class.

**Comment by eaddingtonwhite** [comment]
@kvcache @cprice404 @danielamiao @ksmotiv8 Where did we land on this? Do we have path forward to unblock this PR?

https://github.com/momentohq/momento-cli/pull/159

I like @kvcache suggestions but feels like a larger refactor for sure then what we have currently in this PR. Are we aligned on target state though? Does someone want to summarize where we landed not totally clear to me.
 
cc/ @WillRInternship

**Comment by eaddingtonwhite** [comment]
@kvcache @danielamiao @cprice404 per slack convos I made follow-up ticket to continue the design discussion for the target state and keep it going.
https://github.com/momentohq/momento-cli/issues/160

Going to merge https://github.com/momentohq/momento-cli/pull/159 pr as is with the following command structure as an interim incremental step towards the target state so that we can get the basic item delete functionality in for our users first and then do a bigger refactor.

```
momento cache set
momento cache get
momento cache delete
momento cache create-cache
momento cache list-caches
momento cache delete-cache
```

cc/ @WillRInternship
